sdk/sdk-api/src/main/java/mil/emp3/api/Overlay.java

```diff
@@ -42,41 +42,65 @@ public Overlay(IGeoContainer geoContainer) {
     }
 
     @Override
-    public void addOverlay(IOverlay overlay, boolean visible)
+    public void addOverlay(IOverlay overlay, boolean visible, Object object)
```

Comment:
Looks like there are a two ways of registering for Camera events.

1) CameraEvent listeners added to a Camera instance: camera.addCameraEventListener(...);
2) CameraEvent listeners added to a Map instance: map.addCameraEventListener(...);

When invoking `camera.apply()`:
* For the first case, it looks straightforward to support passing a _userContext_ object via an overloaded _apply_ method. Is it worth doing?
* For the second case, which is where I think you noticed an issue, it is a _ViewChangeEvent_ which will ultimately generate the CameraEvent to notify listeners who registered via the Map instance. The core of the issue is that the _ViewChangeEvent_ gets fired only after the following method call into the WW API, which does not allow us to supply a _userContext_ object: [Emp3NavigationListener.this.ww.requestRedraw();](https://github.com/missioncommand/emp3-android/blob/development/mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/Emp3NavigationListener.java#L359).